File: app/src/main/java/com/elishaazaria/sayboard/AppCtx.java
package com.elishaazaria.sayboard;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.Resources;

import androidx.annotation.ArrayRes;
import androidx.annotation.BoolRes;
import androidx.annotation.IntegerRes;
import androidx.annotation.StringRes;

public class AppCtx {
    @SuppressLint("StaticFieldLeak") // App context
    private static Context appCtx;

    public static void setAppCtx(Context context) {
        appCtx = context.getApplicationContext();
    }

    public static Context getAppCtx() {
        return appCtx;
    }

    public static String getStringRes(@StringRes int res) {
        return appCtx.getString(res);
    }

    public static int getIntegerRes(@IntegerRes int res) {
        return appCtx.getResources().getInteger(res);
    }

    public static boolean getBoolRes(@BoolRes int res) {
        return appCtx.getResources().getBoolean(res);
    }

    public static String[] getStringArrayRes(@ArrayRes int res){
        return appCtx.getResources().getStringArray(res);
    }
}

File: app/src/main/java/com/elishaazaria/sayboard/AppCtx.kt
package com.elishaazaria.sayboard

import android.annotation.SuppressLint
import android.content.Context
import androidx.annotation.ArrayRes
import androidx.annotation.BoolRes
import androidx.annotation.IntegerRes
import androidx.annotation.StringRes

@SuppressLint("StaticFieldLeak")
object AppCtx {
    @JvmStatic
    @SuppressLint("StaticFieldLeak") // App context
    var appCtx: Context? = null
        private set

    @JvmStatic
    fun setAppCtx(context: Context) {
        appCtx = context.applicationContext
    }

    @JvmStatic
    fun getStringRes(@StringRes res: Int): String {
        return appCtx!!.getString(res)
    }

    @JvmStatic
    fun getIntegerRes(@IntegerRes res: Int): Int {
        return appCtx!!.resources.getInteger(res)
    }

    @JvmStatic
    fun getBoolRes(@BoolRes res: Int): Boolean {
        return appCtx!!.resources.getBoolean(res)
    }

    fun getStringArrayRes(@ArrayRes res: Int): Array<String> {
        return appCtx!!.resources.getStringArray(res)
    }
}